### PR TITLE
Feature/angular update

### PR DESCRIPTION
**Pull request by danoswaltCL**
Angular 13 update:

- Updates Angular and Material UI and most dependencies to latest Angular 13 (LTS) versions
- Updates Jest config to be able to understand latest module types
- Fixes some errors that popped up due to more strict state immutability in NGRX library. State that is selected from the Store can't ever ever ever be assigned or modified, you have to clone the object and then update the Store with the new object; there were 4-5 places where this needed fixed.
- Removes some dead dependencies

**NOTE**
This also contains a bump from Node 16.3.0 -> 16.15.0 for docker, as Angular CLI needs 16.15.0 or higher.

Keep an eye out for browser runtime errors that say something like 'cannot assign to read-only property of....', this indicates a place where a JavaScript object has been "frozen" and cannot be modified, likely by NGRX on state that has been read from the store. I tried to find these errors on my own, but TypeScript isn't picking up on them in the code itself for some reason.

Most code changes were minor, but some of the fixes mentioned above may cause code conflicts, I can help resolve. Otherwise it would very good to get this in (putting in draft status for now until others can verify it is working on their machines).

### CODE REVIEW

**Comment by danoswaltCL** [comment]
@jreddig let's hold off on merging this until we can test @VivekFitkariwala 's PR, this one also touches a lot of files. @ppratikcr7 how do you feel about being able to merge this in with your other frontend changes going on?

**Comment by ppratikcr7** [comment]
> @jreddig let's hold off on merging this until we can test @VivekFitkariwala 's PR, this one also touches a lot of files. @ppratikcr7 how do you feel about being able to merge this in with your other frontend changes going on?

Okay, I will merge it in a day or two after my metrics stepper tab is ready.

**Review by jreddig — APPROVED** [review]
Looks good to me!

**Pull request merged**
